py4DSTEM/process/diskdetection/diskdetection_aiml.py

```diff
@@ -0,0 +1,717 @@
+            minIntensity should be an int.
+        name (str): name for the returned PointListArray
+        filter_function (callable): filtering function to apply to each diffraction
+            pattern before peakfinding. Must be a function of only one argument (the
+            diffraction pattern) and return the filtered diffraction pattern. The
+            shape of the returned DP must match the shape of the probe kernel (but does
+            not need to match the shape of the input diffraction pattern, e.g. the filter
+            can be used to bin the diffraction pattern). If using distributed disk
+            detection, the function must be able to be pickled with by dill.
+        _qt_progress_bar (QProgressBar instance): used only by the GUI.
+        model_path (str): filepath for the model weights (Tensorflow model) to load from.
+            By default, if the model_path is not provided, py4DSTEM will search for the 
+            latest model stored on cloud using metadata json file. It is not recommended to
+            keep track of the model path and advised to keep this argument unchanged (None)
+            to always search for the latest updated training model weights.
+            
+    Returns:
+        (PointListArray): the Bragg peak positions and correlation intensities
+    """
+
+    # Make the peaks PointListArray
+    coords = [('qx',float),('qy',float),('intensity',float)]
+    peaks = PointListArray(coordinates=coords, shape=(datacube.R_Nx, datacube.R_Ny))
+
+    # check that the filtered DP is the right size for the probe kernel:
+    if filter_function: assert callable(filter_function), "filter_function must be callable"
+    DP = datacube.data[0,0,:,:] if filter_function is None else filter_function(datacube.data[0,0,:,:])
+    #assert np.all(DP.shape == probe.shape), 'Probe kernel shape must match filtered DP shape'
+    
+    if predict:
+        probe = tf.expand_dims(tf.repeat(tf.expand_dims(probe, axis=0), 
+                                             datacube.R_Nx*datacube.R_Ny, axis=0), axis=-1)
+        DP = tf.expand_dims(tf.reshape(datacube.data,
+                                      (datacube.R_Nx*datacube.R_Ny,datacube.Q_Nx,datacube.Q_Ny)), axis = -1)
+            
+        prediction = np.zeros(shape = (datacube.R_Nx*datacube.R_Ny, datacube.Q_Nx, datacube.Q_Ny, 1))
+        model = _get_latest_model(model_path = model_path)
+        for att in range(num_attmpts):
+            print('attempt {} \n'.format(att+1))
+            prediction += model.predict([DP,probe], batch_size = batch_size, verbose=1)
+        print('Averaging over {} attempts \n'.format(num_attmpts))
+        prediction = prediction/num_attmpts
+    
+    prediction = np.reshape(np.transpose(prediction, (0,3,1,2)),
+                            (datacube.R_Nx, datacube.R_Ny, datacube.Q_Nx, datacube.Q_Ny))
+    
+    if _qt_progress_bar is not None:
+        from PyQt5.QtWidgets import QApplication
```

Comment:
This is indeed already in the main branch Bragg disk detection functions. The `_qt_progress_bar` argument is only meant to be used by the GUI, which needs this little snippet so that the progress bar there gets updated. This snippet also prevents the GUI from becoming locked up while in the middle of doing the disk detection, since otherwise Qt events cannot be handled until the disk detection returns.

Reply:
shall I remove this?